cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -472,10 +472,10 @@ public boolean apply(ProgramRuntimeService.RuntimeInfo runtimeInfo) {
   private void deleteMetrics(String namespaceId, String applicationId) throws Exception {
     Collection<ApplicationSpecification> applications = Lists.newArrayList();
     if (applicationId == null) {
-      applications = this.store.getAllApplications(new Id.Namespace(namespaceId));
+      applications = this.store.getAllApplications(new NamespaceId(namespaceId));
```

Comment:
It would be better to create a variable ``NamespaceId namespace = new NamespaceId(namespaceId)`` outside the ``if``. Then, line 477 can be ``ApplicationSpecification spec = store.getApplication(namespace.app(applicationId``. You can then fix the weird indentation on line 477-478